src/Elastic.Apm.Azure.Functions/ElasticApmMiddleware.cs

```diff
@@ -0,0 +1,126 @@
+		FaasIdPrefix =
+			$"/subscriptions/{metaData?.Account?.Id}/resourceGroups/{metaData?.Project}/providers/Microsoft.Web/sites/{metaData?.Instance}/functions/";
+		Logger.Trace()?.Log($"FaasIdPrefix: {FaasIdPrefix}");
+	}
+
+	private static bool IsColdStart() => Interlocked.Exchange(ref ColdStart, 0) == 1;
+
+	public async Task Invoke(FunctionContext context, FunctionExecutionDelegate next)
+	{
+		try
+		{
+			Logger.Trace()?.Log($"{nameof(Invoke)} - {context.FunctionDefinition.Name}");
+
+			var data = GetTriggerSpecificData(context);
+			await Agent.Tracer.CaptureTransaction(data.Name, ApiConstants.TypeRequest, async t =>
+			{
+				if (t is Transaction transaction) // TODO: Expose FaaS property on ITransaction?
+				{
+					transaction.FaaS = new Faas
+					{
+						Name = context.FunctionDefinition.Name,
+						Id = $"{FaasIdPrefix}{context.FunctionDefinition.Name}",
+						Trigger = new Trigger { Type = data.TriggerType },
+						Execution = context.InvocationId,
+						ColdStart = IsColdStart()
+					};
+				}
+
+				await next(context);
+
+				t.Outcome = Outcome.Success;
+				t.Result = GetTriggerSpecificResult(context);
+			}, data.TracingData);
+		}
+		catch (Exception ex)
+		{
+			Logger.Log(LogLevel.Error, $"Exception was thrown during '{nameof(Invoke)}'", ex, null);
+		}
+	}
+
+	private TriggerSpecificData GetTriggerSpecificData(FunctionContext context)
+	{
+		var httpRequestData = GetHttpRequestData(context);
+		if (httpRequestData != null)
+		{
+			Logger.Trace()?.Log("HTTP Trigger type detected.");
+
+			httpRequestData.Headers.TryGetValues("traceparent", out var traceparent);
+			httpRequestData.Headers.TryGetValues("tracestate", out var tracestate);
+			var name = $"{httpRequestData.Method} {httpRequestData.Url.AbsolutePath}";
+			return new TriggerSpecificData(name, Trigger.TypeHttp,
+				TraceContext.TryExtractTracingData(traceparent?.FirstOrDefault(), tracestate?.FirstOrDefault()));
+		}
+
+		return new TriggerSpecificData(context.FunctionDefinition.Name, Trigger.TypeOther, null);
+	}
+
+	private static string GetTriggerSpecificResult(FunctionContext context)
+	{
+		var httpResponseData = GetProperty<HttpResponseData>(context, "InvocationResult");
+		return httpResponseData != null
+			? $"HTTP {(int)httpResponseData.StatusCode / 100}xx"
+			: "success";
+	}
+
+	private static HttpRequestData? GetHttpRequestData(FunctionContext functionContext)
+	{
+		var inputData = GetProperty<IReadOnlyDictionary<string, object>>(functionContext, "InputData");
+		return inputData?.Values.SingleOrDefault(o => o is HttpRequestData) as HttpRequestData;
+	}
+
+	private static T? GetProperty<T>(FunctionContext functionContext, string name) where T : class
+	{
+		var feature = functionContext.Features.SingleOrDefault(f => f.Key.Name == "IFunctionBindingsFeature").Value;
+		return feature?.GetType().GetProperties().SingleOrDefault(p => p.Name == name)?.GetValue(feature) as T;
```

Comment:
Rider suggests me that you can remove conditional access on `feature?.`:

<img width="1117" alt="image" src="https://user-images.githubusercontent.com/1091853/202558130-2050a12a-675b-417a-b889-c2a443aa202a.png">

Which I don't really get.. `feature` is `object?`, so it can be null.
Although `.Value` in the previous line returns `TValue` and not `TValue?`, so I don't know why Rider thinks it's `object?` and not `object`.

Anyways, I'm confused by Rider, and I think the code is ok. I just wanted to let you know.

Reply:
Hmm, this seems like a Rider issue to me.
The call to `functionContext.Features.SingleOrDefault` returns a `KeyValuePair<Type, object>` which is a `struct`, hence never `null`. The `Value` of this `KeyValuePair` is of type `object` though (the `feature`) variable, so the `check` is required.

Reply:
> Hmm, this seems like a Rider issue to me.

Yeah, I also think that.